### PR TITLE
Adding sampling method from PySMO

### PR DESCRIPTION
**Pull request by eyoung55**
## Fixes/Addresses:
This change pulls in the PySMO LatinHypercubeSampling (LHS) class to allow the parameter sweep tool to test samples which more completely and efficiently explore the parameter space vs. purely random sampling methods.

## Summary/Motivation:
Using the current random sampling methods can produce clumped points which don't sufficiently explore the parameter space.  Incorporating LHS from PySMO will help alleviate this potential issue.

## Changes proposed in this PR:
- Add `LatinHypercubeSample` class
- Added LHS sampling type to trigger `sampling.LatinHypercubeSampling` from PySMO

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by bknueven** [comment]
I like the further integration with existing IDAES tools. It looks like it would be straightforward to pull in other sampling methods from PySMO.

**Comment by eyoung55** [comment]
> It looks like it would be straightforward to pull in other sampling methods from PySMO.

Agreed, PySMO's `UniformSampling` could be used in place of the parameter sweep's `FixedSample` as a near drop-in replacement along the lines of this PR, not to mention the other sampling classes.  Probably makes sense to tackle those in a future PR where.

**Comment by bknueven** [comment]
This is now awaiting #216.

File: proteuslib/tools/parameter_sweep.py [hunk 1]
```diff
@@ -95,6 +98,18 @@ def setup(self, mean, sd):
 
 # ================================================================
 
+class LatinHypercubeSample(UniformSample):
```
Comment (bknueven) [inline proteuslib/tools/parameter_sweep.py:101]:
```suggestion
class LatinHypercubeSample(_Sample):
    sampling_type = SamplingType.RANDOM_LHS
```

File: proteuslib/tools/parameter_sweep.py [hunk 2]
```diff
@@ -95,6 +98,18 @@ def setup(self, mean, sd):
...
+
+    def sample(self, num_samples): 
+        return [self.lower_limit, self.upper_limit]
+
+    def setup(self, lower_limit, upper_limit):
+        self.lower_limit = lower_limit
+        self.upper_limit = upper_limit
+        self.sampling_type = SamplingType.RANDOM_LHS
```
Comment (bknueven) [inline proteuslib/tools/parameter_sweep.py:109]:
```suggestion
```

File: proteuslib/tools/parameter_sweep.py [hunk 3]
```diff
@@ -131,6 +146,15 @@ def _build_combinations(d, sampling_type, num_samples, comm, rank, num_procs):
             sorting = np.argsort(param_values[0])
             global_combo_array = np.vstack(param_values).T
             global_combo_array = global_combo_array[sorting, :]
+
+        elif sampling_type == SamplingType.RANDOM_LHS:
+            lb = [val[0] for val in param_values]
+            ub = [val[1] for val in param_values]
+            lhs = sampling.LatinHypercubeSampling([lb, ub], number_of_samples=num_samples, sampling_type='creation')
```
Comment (bknueven) [inline proteuslib/tools/parameter_sweep.py:153]: If we add more of the PySMO sampling methods, we should consider if there's a more general approach.

**Review by bknueven — COMMENTED** [review]
Looks pretty good. I left a few minor comments.

**Review by bknueven — APPROVED** [review]

**Review by jefalon — APPROVED** [review]

**Pull request merged**